docs/UserGuide.md

```diff
@@ -147,35 +155,56 @@ Examples:
 
 * `tag 1 t/SoftwareEng`
 
-### Searching tasks with name: `find`
+### Searching tasks with name of task: `find`
 
-Searches for a task with a name provided.
+Searches for tasks with a name of task  provided.
 
 Format: `find KEYWORD`
 
+* Searches through ModuleBook 3.5 for tasks whose names contain `KEYWORD`.
+
 Examples:
 
 * `find revise`
 
 ### Searching tasks with tag: `findTag`
 
-Searches for a task with its associated tag.
+Searches for tasks with an associated tag.
 
 Format: `findTag KEYWORD`
 
+* Searches through ModuleBook 3.5 for tasks whose has a tag named `KEYWORD`.
+
 Examples:
 
 * `findTag homework`
 
+### Searching tasks associated to a Module: `mod`
+
+Searches for tasks of a specific Module.
+
+Format: `mod MODULE`
+
+* Searches through ModuleBook 3.5 for tasks which belong to `MODULE`.
+
+Examples:
+
+* `mod CS3243`
+
+
```

Comment:
I see that u have an array of valid modules in ur ModuleManager class... I think it will be good if we specify the valid modules in the UG.. if not, it might be reported as a bug later on haha

Reply:
To add on, maybe the valid module criteria should be listed at the top of the UG, before the instructions. But I think it can be added in later.